File: app/src/main/java/com/example/pic_4_words_java/MainMenu/MainMenuFragmentStateAdapter.java
package com.example.pic_4_words_java.MainMenu;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class MainMenuFragmentStateAdapter extends FragmentStateAdapter {

    public MainMenuFragmentStateAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @Override
    public int getItemCount() {
        return 2;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position){
            case 0:
                return new MainMenuFragment();

            case 1:
                return new LeaderboardFragment();

            default:
                return new MainMenuFragment();
        }
    }

}

File: app/src/main/java/com/example/a4p1w/MainMenuFragmentStateAdapter.kt
package com.example.a4p1w

import  androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.fragment.app.FragmentActivity
import com.example.a4p1w.MainMenu.LeaderboardFragment
import com.example.a4p1w.MainMenu.MainMenuFragment

class MainMenuFragmentStateAdapter(fragmentActivity: FragmentActivity) :FragmentStateAdapter(fragmentActivity) {
    override fun getItemCount(): Int = 2;
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> MainMenuFragment()
            1 -> LeaderboardFragment()
            else -> MainMenuFragment()
        }
    }
}

//The createFragment() method returns the Fragment at the specified position.
//So first will be MainMenuFragment and upon swiping left, we will get LeaderboardFragment
//
//The getItemCount() method returns the number of Fragments there are to be displayed, which, in this case, is two.